### PR TITLE
Added XML comments to most types/fields/methods in the TrueCraft.API namespace

### PR DESCRIPTION
**Pull request by mrpimpunicorn**
Tell me if you want any naming, conventions, etc changed.

### CODE REVIEW

File: TrueCraft.API/Configuration.cs [hunk 1]
```diff
@@ -3,8 +3,17 @@
...
+    /// Abstract base class for configurations read from YAML files.
+    /// </summary>
     public abstract class Configuration
     {
+        /// <summary>
+        /// Creates and returns a new configuration read from a YAML file.
+        /// </summary>
+        /// <typeparam name="T">The configuration type.</typeparam>
+        /// <param name="configFileName">The path to the YAML file.</param>
+        /// <returns></returns>
```
Comment (SirCmpwn) [inline TrueCraft.API/Configuration.cs]: Not necessarily YAML. This depends on the implementation of `Configuration`.

Comment (SirCmpwn) [inline TrueCraft.API/Configuration.cs]: Oh, nevermind, I guess this is YAML. Why does the API layer enforce YAML? Weird.

File: TrueCraft.API/Ray.cs [hunk 2]
```diff
@@ -5,12 +5,24 @@
 
 namespace TrueCraft.API
 {
-    // Mostly taken from the MonoXna project, which is licensed under the MIT license
```
Comment (SirCmpwn) [inline TrueCraft.API/Ray.cs]: Why remove this comment?

**Comment by SirCmpwn** [comment]
Looks good, will merge after we figure out the removed comment thing.

File: TrueCraft.API/Ray.cs (hunk 2 above)
Comment (mrpimpunicorn) [inline TrueCraft.API/Ray.cs]: I moved it into the <remarks> tag for the class' XML comment.

Comment (SirCmpwn) [inline TrueCraft.API/Ray.cs]: That comment was not intended to be exported as part of the public API documentation.

Comment (mrpimpunicorn) [inline TrueCraft.API/Ray.cs]: Okay, I'll remove the remark tag around them so they aren't visible in Intellisense.

Comment (SirCmpwn) [inline TrueCraft.API/Ray.cs]: Can you return it to a normal (`//`) comment at the top of the class?

Comment (mrpimpunicorn) [inline TrueCraft.API/Ray.cs]: Done, is it possible to put the commit with this pull request or do I have to open another?

Comment (SirCmpwn) [inline TrueCraft.API/Ray.cs]: You can just push more commits to master and it'll add them to this PR.

**Comment by mrpimpunicorn** [comment]
Okay, I've pushed the commit.

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**